### PR TITLE
Update HTTP->HTTPS when both URLs work

### PR DESCRIPTION
**Pull request by fulldecent**
This updates each HTTP links to HTTPS if and only if the updated URL is accessible.

We should update our links to HTTPS because this promotes privacy.

This is a pure-play commit for the above feature and it does not touch anything else.

One reason is that there are so many HTTP URLs is because for a brief time the eips.ethereum.org website was available on HTTP without redirect. Therefore, after this change, I expect in the future there will be much less need for changes like the instant PR.

### CODE REVIEW

File: EIPS/eip-1.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Each status change is requested by the EIP author and reviewed by the EIP editor
 * **Draft** -- Once the first draft has been merged, you may submit follow-up pull requests with further changes to your draft until such point as you believe the EIP to be mature and ready to proceed to the next status. An EIP in draft status must be implemented to be considered for promotion to the next status (ignore this requirement for core EIPs).
   * :arrow_right: Last Call -- If agreeable, the EIP editor will assign Last Call status and set a review end date (`review-period-end`), normally 14 days later.
   * :x: Last Call -- A request for Last Call status will be denied if material changes are still expected to be made to the draft. We hope that EIPs only enter Last Call once, so as to avoid unnecessary noise on the RSS feed.
-* **Last Call** -- This EIP will listed prominently on the http://eips.ethereum.org/ website (subscribe via RSS at [last-call.xml](/last-call.xml)).
+* **Last Call** -- This EIP will listed prominently on the https://eips.ethereum.org/ website (subscribe via RSS at [last-call.xml](/last-call.xml)).
```
Comment (jamesray1) [inline EIPS/eip-1.md]: This was here originally but should be changed in a separate PR to
```suggestion
* **Last Call** -- This EIP will be listed prominently on the https://eips.ethereum.org/ website (subscribe via RSS at [last-call.xml](/last-call.xml)).
```
I have done too much editing and other work for Ethereum without getting remunerated for it so can't do that further.

File: EIPS/eip-1015.md [hunk 2]
```diff
@@ -28,16 +28,16 @@ Moving to PoS has been on the roadmap since day 0 for ethereum, along with a red
...
+[EIP 960](https://eips.ethereum.org/EIPS/eip-969) proposes a change in algorithm to avoid mining being dominated by ASICS. Counter arguments by Phil Daian argue among others than [resisting economies of scale is futile and there might be specific security advantadges to specialized hardware](https://pdaian.com/blog/anti-asic-forks-considered-harmful/). One of the main arguments for PoW mining, even when it doesn't provide security, it is useful as a fair distribution mechanism, that **PoW allows any person with a computer, internet access and electricity to obtain currency without having to deal with government imposed currency controls**.
```
Comment (jamesray1) [inline EIPS/eip-1015.md]:
```suggestion
[EIP 960](https://eips.ethereum.org/EIPS/eip-969) proposes a change in algorithm to avoid mining being dominated by ASICS. Counter arguments by Phil Daian argue among others that [resisting economies of scale is futile and there might be specific security advantadges to specialized hardware](https://pdaian.com/blog/anti-asic-forks-considered-harmful/). One of the main arguments for PoW mining, even when it doesn't provide security, it is useful as a fair distribution mechanism, that **PoW allows any person with a computer, internet access and electricity to obtain currency without having to deal with government imposed currency controls**.
```

Comment (jamesray1) [reply EIPS/eip-1015.md]: Not going to look at the rest of this.

File: EIPS/eip-1.md (hunk 1 above)
Comment (fulldecent) [reply EIPS/eip-1.md]: I will make this change in a separate PR. (To keep this one pure-play.)

No date promised. Will credit you. I have other PRs outstanding against EIP1 already, hoping to resolve them first.

**Comment by expede** [comment]
> EIP 1066 requires approval from one of (@carchrae, @naumenkogs, @expede)
> [...]
> EIP 1444 requires approval from one of (@expede, @jenncoop)

I don't have official approval rights on this repo, but the changes to 1066 & 1444 look good 👍

**Comment by ligi** [comment]
> I don't have official approval rights on this repo, but the changes to 1066 & 1444 look good +1

perhaps it would make sense to modify this bot so a thumbs up also counts as approval if the person does not have the permission to approve

**Comment by Arachnid** [comment]
There's no "approval rights"; anyone can approve a PR. Just go to the code tab, click "Review Changes", "Approve", and "Submit".

**Comment by nicksavers** [comment]
I agree with this PR but currently has conflicts. @fulldecent

**Comment by fulldecent** [comment]
@nicksavers thank you, de-conflicted

**Comment by nicksavers** [comment]
I verified that all the updated links are working. Thanks for the work @fulldecent

**Comment by fulldecent** [comment]
Thanks for the review!

**Comment by Dukdig2325** [comment]
> This updates each HTTP links to HTTPS if and only if the updated URL is accessible.
> 
> We should update our links to HTTPS because this promotes privacy.
> 
> This is a pure-play commit for the above feature and it does not touch anything else.
> 
> One reason is that there are so many HTTP URLs is because for a brief time the eips.ethereum.org website was available on HTTP without redirect. Therefore, after this change, I expect in the future there will be much less need for changes like the instant PR.
